app/controllers/SndContactEmailController.scala

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import controllers.actions._
+import exceptions.SomeInformationIsMissingException
+import forms.SndContactEmailFormProvider
+
+import javax.inject.Inject
+import models.Mode
+import models.requests.DataRequest
+import navigation.Navigator
+import pages.{ContactNamePage, SndContactEmailPage, SndContactNamePage}
+import play.api.data.Form
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.libs.json.Json
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import play.twirl.api.Html
+import renderer.Renderer
+import repositories.SessionRepository
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import uk.gov.hmrc.viewmodels.NunjucksSupport
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class SndContactEmailController @Inject() (
+  override val messagesApi: MessagesApi,
+  sessionRepository: SessionRepository,
+  navigator: Navigator,
+  identify: IdentifierAction,
+  getData: DataRetrievalAction,
+  requireData: DataRequiredAction,
+  formProvider: SndContactEmailFormProvider,
+  val controllerComponents: MessagesControllerComponents,
+  renderer: Renderer
+)(implicit ec: ExecutionContext)
+    extends FrontendBaseController
+    with I18nSupport
+    with NunjucksSupport {
+
+  private val form = formProvider()
+
+  private def render(mode: Mode, form: Form[String])(implicit request: DataRequest[AnyContent]): Future[Html] = {
+    val data = Json.obj(
+      "form"   -> form,
+      "name"   -> request.userAnswers.get(SndContactNamePage).getOrElse(throw new SomeInformationIsMissingException("Missing contact name")),
```

Comment:
@iservport @mbiczak Rather than throwing an exception, let's handle it gracefully  if name is None redirect to Missing Information page using a match case

Reply:
Jira ticket https://jira.tools.tax.service.gov.uk/browse/DAC6-1040 created to address the comment